pyphare/pyphare/pharesee/geometry.py

```diff
@@ -0,0 +1,277 @@
+from ..core import box as boxm
+from .hierarchy import FieldData, is_root_lvl
+
+
+
+def toFieldBox(box, patch_data):
+    """
+    grows the box by one cell if patch_data is primal
+    because box.upper+1 allows to get the rightmost primal node
+    otherwise missing
+    """
+    assert (isinstance(patch_data, FieldData))
+    if patch_data.layout.centering["X"][patch_data.field_name] == "primal":
+        return boxm.Box(box.lower, box.upper + 1)
+    else:
+        return box
+
+
+
+
+
+def shift_patch(patch, offset):
+    patch.box = boxm.shift(patch.box, offset)
+    for pdata in patch.patch_datas.values():
+        pdata.ghost_box = boxm.shift(pdata.ghost_box, offset)
+
+
+
+
+def touch_domain_border(box, domain_box, border):
+    if border == "upper":
+        return domain_box.upper in box
+    elif border == "lower":
+        return domain_box.lower in box
+    else:
+        raise RuntimeError("invalid border")
+
+
+
+def compute_overlaps(patches, domain_box):
+    """
+    returns a list of overlaps for all patch datas in given patches
+    and for a domain box. An overlap is defined as an intersection of
+    two PatchData ghost box.
+
+    if  - = domain cell, o = ghost cell
+
+    PatchData1  : - - - - - - - - - o o o o o
+    PatchData2  :                     o o o o o - - - - - - - -
+    overlap box :                     ^ ^ ^ ^
+
```

Comment:
this is not (only) about contiguous patches. The goal here is to return all the overlaps existing in a collection of patches. An overlap is defined, as the drawing shows, as a box composed of cells existing in two patchdatas, or said differently, as the box resulting from the intersection of two patch data ghost boxes.
is that clearer?

Reply:
yes it is... so the overlap is only for the ghost cells ?
no overlap if it only concerns domain cells and not ghost cells.

Reply:
no, an overlap concerns all cells. 
I think you're confused because your forgot what a ghost box is. A ghost box is a box grown by the ghost area.
if the box is (0,32) and there is 1 ghost cell, the ghost box is (-1,33).
the overlap is just the intersection of two ghost boxes. nothing more.

so if you have (0,32) and (33,64) for boxes, and ghost cell width is 5, you have ghost boxes (-5,37) and (28,69) and the overlap is (28,37), that's simple.